### PR TITLE
Added tests for Button component

### PR DESCRIPTION
**Pull request by jacobsimon**
Added some unit tests for the button component using [enzyme](https://github.com/airbnb/enzyme) (thanks for the suggestion @osdiab!) Doesn't cover every case but most of the basic functionality.

### CODE REVIEW

File: test/Button_test.jsx [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+      it(`renders a ${size}, ${type} button with the correct classes`, () => {
+        const button = shallow(<Button size={size} type={type} value="A button" />);
+        assert(button.hasClass("Button"));
+        assert(button.hasClass(`Button--${size}`));
+        assert(button.hasClass(`Button--${type}`));
+      });
+    });
+  });
+
+  it("throws an error for small, destructive buttons", () => {
```
Comment (osdiab) [inline test/Button_test.jsx]: How about the other errors?

File: test/Button_test.jsx [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+    assert.equal(button.text(), "A submit button");
+  });
+
+  it("renders a <a> element if href is provided", () => {
+    const button = shallow(<Button value="A link" href="http://clever.com" />);
+    assert.equal(button.find("a[target='_blank']").length, 1);
+    assert.equal(button.text(), "A link");
+  });
+
+  it("calls the onClick handler when clicked", () => {
```
Comment (osdiab) [inline test/Button_test.jsx]: Pretty :)

**Comment by osdiab** [comment]
Aside from missing some error state checks, LGTM!

File: test/Button_test.jsx (hunk 1 above)
Comment (jacobsimon) [inline test/Button_test.jsx]: https://github.com/Clever/components/pull/18/commits/6d086cd39e1b68d1018ce338f14302795643a150

File: test/Button_test.jsx [hunk 3]
```diff
@@ -0,0 +1,68 @@
+/* eslint func-names: "off" */
```
Comment (rgarcia) [inline test/Button_test.jsx]: @jacobsimon afaict the tests pass without this exception?

**Pull request merged**